scalding-core/src/main/scala/com/twitter/scalding/typed/KeyGrouping.scala

```diff
@@ -0,0 +1,25 @@
+package com.twitter.scalding.typed
+
+import com.twitter.scalding.serialization.{
+  OrderedSerialization,
+  OrderedSerialization2
+}
+import scala.math.Ordering
+
+case class KeyGrouping[T](ord: Ordering[T])
```

Comment:
If we introduce a new type here I’d like to be a sealed trait and either be an Ordering or an OrderedSerialization.